### PR TITLE
1579886 - Add iOS lifecycle observer

### PR DESCRIPTION
**Pull request by travis79**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/ios/Glean.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		1F60589723148BF800307A9F /* GleanLifecycleObserver.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1F60589623148BF800307A9F /* GleanLifecycleObserver.swift */; };
```
Comment (badboy) [inline glean-core/ios/Glean.xcodeproj/project.pbxproj]: I'm already getting annoyed by the fact that every PR will change the project file now ... (not your fault of course)

File: glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+class GleanLifecycleObserver {
```
Comment (Dexterp37) [inline glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift]: nit: should we have docs for this? Will `GleanLifecycleObserver` be part of the API surface?

Comment (travis79) [reply glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift]: I don't think it should show up as part of the public API, according to [documentation](https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html) the default access in Swift is internal. You have to be explicit about public access, so it was not my intent to allow this to leak to the public API.  (Unless you see a need to, that is)

File: glean-core/ios/Glean.xcodeproj/project.pbxproj (hunk 1 above)
Comment (travis79) [reply glean-core/ios/Glean.xcodeproj/project.pbxproj]: Yeah, it's annoying to be sure.  It's definitely not as clean as the Android approach to this.

File: glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift (hunk 2 above)
Comment (travis79) [reply glean-core/ios/Glean/Scheduler/GleanLifecycleObserver.swift]: If you think this should be public, we can file a follow up bug to this.

**Pull request merged**